compiler/optimizer/LocalOpts.cpp

```diff
@@ -7594,6 +7594,9 @@ int32_t TR_InvariantArgumentPreexistence::perform()
 static bool specializeInvokeExactSymbol(TR::Node *callNode, TR::KnownObjectTable::Index receiverIndex, TR::Compilation *comp, TR::Optimization *opt)
```

Comment:
The check for the knot is valid, but surely there needs to be some logic to stop calling this if you don't have a valid knot index for the receiverIndex - we'd just be wasting compile time otherwise...

Reply:
I noticed the place where this is called is guarded by `receiverInfo.hasKnownObjectIndex()` so perhaps checking for knot here is unnecessary.

Reply:
Commit amended